cli/command/load_balancer.go

```diff
@@ -10,6 +17,449 @@ func NewLoadBalancerCommand(b *CommandBase) *LoadBalancerCommand {
+				Usage:     "List all load balancers",
+				Action:    l.list,
+				ArgsUsage: " ",
+			},
+		},
+	}
+}
+
+func (l *LoadBalancerCommand) addport(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "LOADBALANCER_NAME", "PORT")
+	if err != nil {
+		return err
+	}
+
+	port, err := parsePort(args["PORT"], c.String("certificate"))
+	if err != nil {
+		return err
+	}
+
+	loadBalancerID, err := l.resolveSingleEntityIDHelper("load_balancer", args["LOADBALANCER_NAME"])
+	if err != nil {
+		return err
+	}
+
+	loadBalancer, err := l.client.ReadLoadBalancer(loadBalancerID)
+	if err != nil {
+		return err
+	}
+
+	ports := append(loadBalancer.Ports, *port)
+	req := models.UpdateLoadBalancerRequest{
+		LoadBalancerID: loadBalancerID,
+		Ports:          &ports,
+	}
+
+	jobID, err := l.client.UpdateLoadBalancer(req)
+	if err != nil {
+		return err
+	}
+
+	onCompleteFn := func(loadBalancerID string) error {
+		loadBalancer, err := l.client.ReadLoadBalancer(loadBalancerID)
+		if err != nil {
+			return err
+		}
+
+		l.printer.StopSpinner()
+		return l.printer.PrintLoadBalancers(loadBalancer)
+	}
+
+	return l.waitOnJobHelper(c, jobID, "Adding port to load balancer", onCompleteFn)
+}
+
+func (l *LoadBalancerCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT_TARGET", "LOADBALANCER_NAME")
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := l.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT_TARGET"])
+	if err != nil {
+		return err
+	}
+
+	ports := []models.Port{}
+	for _, p := range c.StringSlice("port") {
+		port, err := parsePort(p, c.String("certificate"))
+		if err != nil {
+			return err
+		}
+
+		ports = append(ports, *port)
+	}
+
+	if c.String("healthcheck-target") != "" {
+		if _, err := parseTarget(c.String("healthcheck-target")); err != nil {
+			return err
+		}
+	}
+
+	healthCheck := models.HealthCheck{
+		Target:             c.String("healthcheck-target"),
+		Interval:           c.Int("healthcheck-interval"),
+		Timeout:            c.Int("healthcheck-timeout"),
+		HealthyThreshold:   c.Int("healthcheck-healthy-threshold"),
+		UnhealthyThreshold: c.Int("healthcheck-unhealthy-threshold"),
+	}
+
+	req := models.CreateLoadBalancerRequest{
+		LoadBalancerName: args["LOADBALANCER_NAME"],
+		EnvironmentID:    environmentID,
+		IsPublic:         !c.Bool("private"),
+		Ports:            ports,
+		HealthCheck:      healthCheck,
+	}
+
+	jobID, err := l.client.CreateLoadBalancer(req)
+	if err != nil {
+		return err
+	}
+
+	onCompleteFn := func(loadBalancerID string) error {
+		loadBalancer, err := l.client.ReadLoadBalancer(loadBalancerID)
+		if err != nil {
+			return err
+		}
+
+		l.printer.StopSpinner()
+		return l.printer.PrintLoadBalancers(loadBalancer)
+	}
+
+	return l.waitOnJobHelper(c, jobID, "Creating Load Balancer", onCompleteFn)
+}
+
+func (l *LoadBalancerCommand) delete(c *cli.Context) error {
+	return l.deleteHelper(c, "loadbalancer", func(loadBalancerID string) (string, error) {
```

Comment:
the `type` needs to be `load_balancer` when resolving

Reply:
:white_check_mark: